### PR TITLE
8345656: Move os alignment functions out of ReservedSpace

### PR DESCRIPTION
**Pull request by stefank**
I have a wish to simplify the ReservedSpace classes (See: [JDK-8345655](https://bugs.openjdk.org/browse/JDK-8345655)) and as a small step I would like to move the small helper functions that align addresses and sizes to `os::vm_page_size()` and `os::vm_allocation_granularity()` out to be `os::` helpers.

Tested tier1-3 together with the other patches in the JDK-8345655 prototype.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345656](https://bugs.openjdk.org/browse/JDK-8345656): Move os alignment functions out of ReservedSpace (**Enhancement** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22600/head:pull/22600` \
`$ git checkout pull/22600`

Update a local copy of the PR: \
`$ git checkout pull/22600` \
`$ git pull https://git.openjdk.org/jdk.git pull/22600/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22600`

View PR using the GUI difftool: \
`$ git pr show -t 22600`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22600.diff">https://git.openjdk.org/jdk/pull/22600.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22600#issuecomment-2522718847)
</details>

### CODE REVIEW

**Review by afshin-zafari — COMMENTED** [review]
Thanks for cleaning up the Reserved Space stuff. I found that Copyright years are not up-to-date for these files:
```
gc/g1/g1CardTable.cpp
gc/g1/g1CardTable.hpp
gc/shared/markBitMap.cpp
gc/shenandoah/shenandoahMarkBitMap.cpp

```

**Comment by stefank** [comment]
Thanks @afshin-zafari for the review!

File: src/hotspot/share/runtime/os.hpp [hunk 1]
```diff
@@ -404,6 +405,9 @@ class os: AllStatic {
   // Return the default page size.
   static size_t vm_page_size() { return OSInfo::vm_page_size(); }
 
+  static size_t align_up_vm_page_size(size_t size)   { return align_up  (size, os::vm_page_size()); }
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/os.hpp:408]: Style Nit: please don't add extra whitespace before the opening function parenthesis.

**Review by dholmes-ora — COMMENTED** [review]
I don't see the removal of the "moved" functions from ./share/memory/virtualspace.cpp ??

**Comment by stefank** [comment]
> I don't see the removal of the "moved" functions from ./share/memory/virtualspace.cpp ??

Good point. That got lost in the split-out from my other rewrites.

File: src/hotspot/share/runtime/os.hpp (hunk 1 above)
Comment (dholmes-ora) [reply src/hotspot/share/runtime/os.hpp:408]: No not a strong request. I see there is precedent for this already in the file - though in some cases it isn't even needed for alignment purposes. I find extra whitespace between a function name and the opening parenthesis to be visually jarring and it makes me have to re-read to ensure I'm not mis-reading something. YMMV.

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks

**Comment by stefank** [comment]
Thanks for the reviews!
/integrate

**Pull request closed without merging**